test/test_child.rb

```diff
@@ -224,6 +224,65 @@ def test_utf8_input_long
     assert p.success?
   end
 
+  def test_streaming_stdout
+    stdout_buf = ""
+    stdout_stream = Proc.new do |chunk|
+      stdout_buf << chunk
+
+      false
+    end
+
+    input = "hello!"
+    p = Child.new('cat', :input => input, :streams => {
+      :stdout => stdout_stream
+    })
+
+    assert p.success?
+    assert_equal input, stdout_buf
+  end
+
+  def test_streaming_stderr
+    stderr_buf = ""
+    stderr_stream = Proc.new do |chunk|
+      stderr_buf << chunk
+
+      false
+    end
+
+    p = Child.new('ls', '-?', :streams => {
+      :stderr => stderr_stream
+    })
+
+    refute p.success?
+    refute stderr_buf.empty?
+  end
+
+  def test_streaming_stdout_aborted
+    stdout_stream = Proc.new do |chunk|
+      true
+    end
+
+    input = "hello!"
+    assert_raises POSIX::Spawn::Aborted do
+      p = Child.new('cat', :input => input, :streams => {
+        :stdout => stdout_stream
+      })
+    end
+  end
+
+  def test_streaming_stderr_aborted
+    stderr_stream = Proc.new do |chunk|
+      true
+    end
+
+    input = "hello!"
+    assert_raises POSIX::Spawn::Aborted do
+      p = Child.new('ls', '-?', :streams => {
+        :stderr => stderr_stream
+      })
+    end
+  end
+
```

Comment:
There are no tests that involve reading more than one `BUFSIZE` worth of output, or reading from both stdout and stderr. Those might be worthwhile additions.